core/src/main/java/org/apache/calcite/sql/dialect/BigQuerySqlDialect.java

```diff
@@ -292,6 +295,17 @@ private static void unparseItem(SqlWriter writer, SqlCall call, int leftPrec) {
     writer.endList(frame);
   }
 
+  private static void unparseWithinGroup(SqlWriter writer, SqlCall call, int leftPrec, int rightPrec) {
+    assert call.operandCount() == 2;
+    call.operand(0).unparse(writer, 0, 0);
+    writer.keyword("OVER");
```

Comment:
so I think you're more familiar with these clauses than me, is it guaranteed that the WITHIN GROUP translates to BQ's OVER in every case? are they really just synonyms for each other, I mean

Reply:
So I detailed in this [comment](https://issues.apache.org/jira/browse/CALCITE-5955?focusedCommentId=17772373&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-17772373) how Calcite currently only supports 
`select percentile_cont(0.5) within group (order by X)`
so I want to say that for what is supported at this moment, it is sufficient

Reply:
Ohhh ok. So if Calcite ended up supporting the `partition by` case (your `input_2` from that comment), do you know if this would break?

Reply:
Julian had a comment [here](https://issues.apache.org/jira/browse/CALCITE-6039) about how we should approach the followup tickets
I think it won't work out of the box, there are things to figure out 
We would still be parsing it as `within_group` in the parser, but want it to be handled differently in the validator.

 
questions to then consider: would we still use a SqlWithinGroupOperator? If yes, then that would be unparsed into "OVER" correctly. If not, this may have to change